.github/workflows/docker.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: Build & Push Docker Images
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+    paths:
+      - scripts/*.dockfile
+      - scripts/*.dockerfile
+      - scripts/setup-*.sh
+      - .github/workflows/docker.yml
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.repository }}-${{ github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      
+      - uses: actions/checkout@v3
+
+      - name: Build and Push check
+        uses: docker/build-push-action@v3
+        with:
+          context: scripts
+          file: scripts/check-container.dockfile
+          build-args: cpu_target=avx
+          push: ${{ github.repository == 'facebookincubator/velox' && github.event != 'pull_request'}}

```

Comment:
I am curious, why do we need to check if repository is velox and event a pull_request ?

Reply:
This will stop the build from failing due to authentication issues (pull request do not have secret access/package write access) and will also avoid failing workflows (and more importantly notification emails) on forks as they of course also do not have access to push to the facebookincubator/velox ghcr repo.